packages/mds-stream-processor/labelers/index.ts

```diff
@@ -0,0 +1,23 @@
+import db from '@mds-core/mds-db'
+import { UUID, VEHICLE_TYPE, PROPULSION_TYPE } from '@mds-core/mds-types'
+
+type MessageLabeler<TMessage, TLabel> = (message: TMessage) => Promise<TLabel>
+
+export interface DeviceLabel {
+  vehicle_type: VEHICLE_TYPE
+  vehicle_propulsion: PROPULSION_TYPE[]
+}
+
+export const deviceLabeler: MessageLabeler<{ device_id: UUID }, DeviceLabel> = async ({ device_id }) => {
+  const { type: vehicle_type, propulsion: vehicle_propulsion } = await db.readDevice(device_id)
+  return { vehicle_type, vehicle_propulsion }
+}
+
+export interface MessageLatencyLabel {
+  message_latency_ms: number
+}
+
+export const messageLatencyLabeler: MessageLabeler<
+  { timestamp: number; recorded: number },
+  MessageLatencyLabel
+> = async ({ timestamp, recorded }) => ({ message_latency_ms: recorded - timestamp })
```

Comment:
Things you pass to `Promise.all` don't _have_ to be _async_ but when defining services/interfaces, I prefer to always make things async from the beginning because it's really hard to re-plumb a synchronous service once it starts getting used.

Reply:
Yup -- makes sense. I knew you could pass a sync method into a Promise.all(), but it feels rather odd to do that.